client-java/instrumentation/src/main/java/org/evomaster/client/java/instrumentation/coverage/methodreplacement/thirdpartyclasses/OpenSearchClientClassReplacement.java

```diff
@@ -0,0 +1,85 @@
+package org.evomaster.client.java.instrumentation.coverage.methodreplacement.thirdpartyclasses;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.Arrays;
+import java.util.List;
+import org.evomaster.client.java.instrumentation.OpenSearchCommand;
+import org.evomaster.client.java.instrumentation.coverage.methodreplacement.Replacement;
+import org.evomaster.client.java.instrumentation.coverage.methodreplacement.ThirdPartyCast;
+import org.evomaster.client.java.instrumentation.coverage.methodreplacement.ThirdPartyMethodReplacementClass;
+import org.evomaster.client.java.instrumentation.coverage.methodreplacement.UsageFilter;
+import org.evomaster.client.java.instrumentation.shared.ReplacementCategory;
+import org.evomaster.client.java.instrumentation.shared.ReplacementType;
+import org.evomaster.client.java.instrumentation.staticstate.ExecutionTracer;
+
+/**
+ * Class responsible to handle method calls to the OpenSearchClient and intercepting
+ * them to include its information (including operation and arguments) on the ExecutionTracer.
+ */
+public class OpenSearchClientClassReplacement extends ThirdPartyMethodReplacementClass {
+    private static final OpenSearchClientClassReplacement singleton = new OpenSearchClientClassReplacement();
+
+    private static final String THIRD_PARTY_CLASS = "org.opensearch.client.opensearch.OpenSearchClient";
+    private static final String GET_METHOD = "get";
+    private static final String SEARCH_METHOD = "search";
+    private static final String INDEX_METHOD = "index";
+
+    @Override
+    protected String getNameOfThirdPartyTargetClass() {
+        return THIRD_PARTY_CLASS;
+    }
+
+    @Replacement(type = ReplacementType.TRACKER, id = GET_METHOD, usageFilter = UsageFilter.ANY, category = ReplacementCategory.OPENSEARCH, castTo = "org.opensearch.client.opensearch.core.GetResponse")
+    public static <TDocument> Object get(Object openSearchClient, @ThirdPartyCast(actualType = "org.opensearch.client.opensearch.core.GetRequest") Object request, Class<TDocument> documentClass) {
+        return handleMethod(openSearchClient, GET_METHOD, Arrays.asList(request, documentClass), request);
+    }
+
+    @Replacement(type = ReplacementType.TRACKER, id = SEARCH_METHOD, usageFilter = UsageFilter.ANY, category = ReplacementCategory.OPENSEARCH, castTo = "org.opensearch.client.opensearch.core.SearchResponse")
+    public static <TDocument> Object search(Object openSearchClient, @ThirdPartyCast(actualType = "org.opensearch.client.opensearch.core.SearchRequest") Object request, Class<TDocument> documentClass) {
```

Comment:
for generic types, better to use single letter, eg `T` instead of `TDocument`, as it makes clear it is a generic and not an existing class